### PR TITLE
fix(api): during promotion, get build module in parrallel

### PR DESCRIPTION
**Pull request by sguiheux**
+ fix buildinfo on non docker artifact: build info missing link to artifact

### CODE REVIEW

File: contrib/integrations/artifactory/artifactory.go [hunk 1]
```diff
@@ -257,100 +258,136 @@ func PrepareBuildInfo(ctx context.Context, artiClient artifact_manager.ArtifactM
...
 	log.Debug(ctx, "retrieve:ModulesFiles repoName:%s path:%s", repoName, path)
+	_, endc := telemetry.Span(ctx, "artifactoryClient.GetFileInfo", telemetry.Tag("path", path), telemetry.Tag("repoName", repoName))
+	fileInfo, err := client.GetFileInfo(repoName, path)
+	endc()
+	if err != nil {
+		return nil, err
+	}
+
+	// If it can be downloaded, it's a file
+	if fileInfo.DownloadURI != "" {
```
Comment (richardlt) [inline contrib/integrations/artifactory/artifactory.go:381]: I think the root path of an artifact is always a folder.

**Review by richardlt — CHANGES_REQUESTED** [review]

File: contrib/integrations/artifactory/artifactory.go (hunk 1 above)
Comment (sguiheux) [reply contrib/integrations/artifactory/artifactory.go:381]: no, only for docker

File: contrib/integrations/artifactory/artifactory.go [hunk 2]
```diff
@@ -257,100 +258,138 @@ func PrepareBuildInfo(ctx context.Context, artiClient artifact_manager.ArtifactM
...
+	}
+	switch data.RepoType {
+	case "docker":
+		mod.Type = buildinfo.Docker
+		props := make(map[string]string)
+		parsedUrl, err := url.Parse(client.GetURL())
 		if err != nil {
 			endc()
-			return nil, err
+			return nil, sdk.WrapError(err, "unable to parse artifactory url [%s]: %v", client.GetURL())
```
Comment (fsamin) [inline contrib/integrations/artifactory/artifactory.go:335]: missing argument or invalid log message

**Review by fsamin — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18589.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18589.2 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18589.2 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ■

**Review by fsamin — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18589.3 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18589.4 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

**Review by richardlt — APPROVED** [review]

**Pull request merged**